src/actions/transferActions.js

```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
+      let matchedOutput = txData.outputs.find(output => output.address === account.address)
+      if (matchedInput) {
+        // if address is in inputs, then it is transfering out
+        //
+        // the destination address is assumed to be the first one in the outputs which
+        // is not account.address
+        // note that having multiple output addresses is possible
+
+        let _out = txData.outputs.find(
+          output => output.address && output.address !== account.address
+        )
+        from = account.address
+        to = _out.address
+        amount = _out.value
+      } else if (matchedOutput) {
+        // if address only exists in outputs, then it is transfering in
+        from = matchedOutput.address
+        to = account.address
+        amount = matchedOutput.value
+      } else {
+        // this should not happen
+        throw new Error(`Cannot find matched input or output for ${txHash}`)
+      }
+
+      // fill in data
+      rv[txHash] = {
+        timestamp: moment(txData.received_at).unix(),
+        blockNumber: txData.block.height,
+        txFees: txData.fees
+      }
+      rv[txHash].standaloneTx = { from, to, value: amount, cryptoType: account.cryptoType }
+    }
+  } else if (getCryptoPlatformType(account.cryptoType) === 'ethereum') {
+    // address is in lowercase in returned txData
+    // must convert account address to lowercase first
+    const accountAddr = account.address.toLowerCase()
+
+    for (let result of unmatched) {
+      let txHash = result.txHash
+      let txData = txs[txHash2Idx[txHash]]
+      if (isERC20(account.cryptoType)) {
+        // token tx
+        const tokenContractAddr = getCrypto(account.cryptoType).address.toLowerCase()
+        let {
+          contractAddress,
+          from,
+          to,
+          value,
+          timeStamp,
+          blockNumber,
+          nonce,
+          gasPrice,
+          cumulativeGasUsed
+        } = txData
+
+        if (contractAddress === tokenContractAddr && (from === accountAddr || to === accountAddr)) {
+          // token matches
+          // fill in data
+          rv[txHash] = { timestamp: timeStamp, blockNumber, nonce, gasPrice, cumulativeGasUsed }
+          rv[txHash].standaloneTx = {
+            from,
+            to,
+            value,
+            cryptoType: account.cryptoType
+          }
+        }
+      } else {
+        // eth tx
+        let { from, to, value, timeStamp, blockNumber, nonce, gasPrice, cumulativeGasUsed } = txData
+        if (from === accountAddr || to === accountAddr) {
+          // from or to matches
+          // fill in data
+          rv[txHash] = { timestamp: timeStamp, blockNumber, nonce, gasPrice, cumulativeGasUsed }
+          rv[txHash].standaloneTx = {
+            from,
+            to,
+            value,
+            cryptoType: account.cryptoType
+          }
+        }
+      }
+    }
+  }
```

Comment:
Put line 749 - 842 input a separate function

Reply:
cannot move due to multiple dependencies